java/core/src/main/java/co/worklytics/psoxy/impl/SanitizerImpl.java

```diff
@@ -65,17 +73,30 @@ List<JsonPath> applicablePaths(@NonNull List<Pair<Pattern, List<JsonPath>>> rule
 
     @Override
     public boolean isAllowed(@NonNull URL url) {
-        if (options.getRules().getAllowedEndpointRegexes() == null
-             || options.getRules().getAllowedEndpointRegexes().isEmpty()) {
-            return true;
+        if (options.getRules2() == null) {
+            if (options.getRules().getAllowedEndpointRegexes() == null
+                || options.getRules().getAllowedEndpointRegexes().isEmpty()) {
+                return true;
+            } else {
+                if (compiledAllowedEndpoints == null) {
+                    compiledAllowedEndpoints = options.getRules().getAllowedEndpointRegexes().stream()
+                        .map(regex -> Pattern.compile(regex, CASE_INSENSITIVE))
+                        .collect(Collectors.toList());
+                }
+                String relativeUrl = URLUtils.relativeURL(url);
+                return compiledAllowedEndpoints.stream().anyMatch(p -> p.matcher(relativeUrl).matches());
+            }
         } else {
             if (compiledAllowedEndpoints == null) {
-                compiledAllowedEndpoints = options.getRules().getAllowedEndpointRegexes().stream()
+                compiledAllowedEndpoints = options.getRules2().getEndpoints().stream()
+                    .map(Rules2.Endpoint::getPathRegex)
                     .map(regex -> Pattern.compile(regex, CASE_INSENSITIVE))
                     .collect(Collectors.toList());
             }
             String relativeUrl = URLUtils.relativeURL(url);
-            return compiledAllowedEndpoints.stream().anyMatch(p -> p.matcher(relativeUrl).matches());
+
+            return options.getRules2().getAllowAllEndpoints() ||
+                compiledAllowedEndpoints.stream().anyMatch(p -> p.matcher(relativeUrl).matches());
```

Comment:
imho, general best practice is to keep constructors as light as possible.  This is "just in time" initialization, which I think of as best practice.

also, exceptions being thrown out of constructors seems very likely to cause problems and not be well-handled, as constructors often invoked by frameworks, such as dagger.  If something blows up in the object graph at run time bc of malformed rules, I think it will be hard to hunt down why